pkg/cloudprovider/gcp.go

```diff
@@ -255,6 +237,28 @@ func (g *GCP) getNetworkInterfaces(instance *google.Instance) ([]*google.Network
 	return instance.NetworkInterfaces, nil
 }
 
+//  This is what the node's providerID looks like on GCP
+// 	spec:
+//   providerID: gce://openshift-gce-devel-ci/us-east1-b/ci-ln-pvr3lyb-f76d1-6w8mm-master-0
+//  i.e: projectID/zone/instanceName
+// split out and return these components
+func splitGCPNode(node *corev1.Node) (project, zone, instance string, err error) {
+	u, err := url.Parse(node.Spec.ProviderID)
+	if err != nil {
+		err = fmt.Errorf("failed to parse node %s provider id %s: %w", node.Name, node.Spec.ProviderID, err)
+		return
+	}
+	project = u.Host
+	parts := strings.SplitN(u.Path, "/", 3)
+	if len(parts) != 3 {
+		err = fmt.Errorf("failed to parse node %s provider id %s: expected two path components", node.Name, node.Spec.ProviderID)
```

Comment:
Could it be that we panic below depending on how long `parts` is once we error here? Might be safer to just put a `return` right after the error

Reply:
oh crap. Forgot the return. Very good catch!!